anchore_engine/services/policy_engine/__init__.py

```diff
@@ -376,15 +330,22 @@ def handle_grypedb_sync(*args, **kwargs):
     cycle_time = kwargs["mythread"]["cycle_timer"]
 
     while True:
-        try:
-            result = GrypeDBSyncTask().execute()
-
-            if result:
-                logger.info("Grype DB synced to local instance via handler")
-        #         TODO narrow scope of exceptions in handlers. see https://github.com/anchore/anchore-engine/issues/1005
-        except Exception:
-            logger.exception(
-                "Error encountered when running GrypeDBSyncTask from async monitor"
+        provider = get_provider(get_vulnerabilities_config())
```

Comment:
I had started doing some work on this and I decided to put this logic in the task. While not needed now, I thought it might be nice to have a handler for the local sync that checks the configuration. If a process wants to check the config before doing the sync, then it could call the task but if it wants to skip that it can just call the sync manager class directly. Thoughts?

Reply:
I would rather keep this out of the task. Tasks by definition are do-this-one-thing-as-you-are-told. The burden of deciding whether to fire a task or not is on the caller.